drivers/sensor/nordic/qdec_nrfx/qdec_nrfx.c

```diff
@@ -272,7 +289,7 @@ static int qdec_nrfx_init(const struct device *dev)
 		.qdec = NRFX_QDEC_INSTANCE(idx),					     \
 		.config = {								     \
 			.reportper = NRF_QDEC_REPORTPER_40,				     \
-			.sampleper = NRF_QDEC_SAMPLEPER_2048US,				     \
+			.sampleper = DT_INST_STRING_TOKEN_OR(idx, nordic_period, idx),       \
```

Comment:
```suggestion
			.sampleper = DT_INST_STRING_TOKEN(idx, nordic_period),		     \
```

Reply:
Right. It should be:
```suggestion
			.sampleper = DT_STRING_TOKEN(QDEC(idx), nordic_period),		     \
```